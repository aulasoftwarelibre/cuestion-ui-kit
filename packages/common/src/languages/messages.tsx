const messages = {
  askQuestion: {
    defaultMessage: "Ask a question",
    description: "",
    id: "askQuestion",
  },
  enterLengthCharacters: {
    defaultMessage: `Enter {length} characters`,
    description: "",
    id: "enterLengthCharacters",
  },
  enterLengthCharactersCode: {
    defaultMessage: `Enter {length} characters code`,
    description: "",
    id: "enterLengthCharactersCode",
  },
  pressEnter: {
    defaultMessage: "and press ENTER.",
    description: "",
    id: "pressEnter",
  },
  send: {
    defaultMessage: "Send",
    description: "",
    id: "send",
  },
  sessionCode: {
    defaultMessage: "Session Code",
    description: "",
    id: "sessionCode",
  },
  yourName: {
    defaultMessage: "Your name (optional)",
    description: "",
    id: "yourName",
  },
};

export default messages;
