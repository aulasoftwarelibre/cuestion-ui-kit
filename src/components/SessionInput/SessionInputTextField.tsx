import TextField from "@material-ui/core/TextField";
import * as React from "react";

interface Props {
    handleOnSubmit: any;
    length: number;
    messageEnter: string;
    messageEnterLengthCharactersCode: string;
    messageSessionCode: string;
}

const SessionInputTextField: React.FunctionComponent<Props> = ({
    handleOnSubmit,
    length,
    messageEnter,
    messageEnterLengthCharactersCode,
    messageSessionCode
}) => {
    const [code, setCode] = React.useState<string>("");

    const handleChange = () => (event: React.ChangeEvent<HTMLInputElement>) => {
        setCode(
            event.target.value
                .toUpperCase()
                .substring(0, length)
                .replace(/[^A-Z0-9]+/g, "")
        );
    };

    const onSubmit = (e: React.FormEvent<HTMLFormElement>) => {
        e.preventDefault();
        const sessionCode: string = e.target[1].value;
        if (sessionCode.length === length) {
            handleOnSubmit(sessionCode);
            setCode("");
        }
    };

    return (
        <form onSubmit={onSubmit}>
            <TextField
                id="session-code-input"
                name="code"
                type="text"
                label={messageSessionCode}
                helperText={
                    code.length === length
                        ? messageEnter
                        : messageEnterLengthCharactersCode
                }
                onChange={handleChange()}
                value={code}
                variant="outlined"
            />
        </form>
    );
};

export default SessionInputTextField;
