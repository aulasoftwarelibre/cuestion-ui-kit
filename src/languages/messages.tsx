const messages = {
  sessionCode: {
    defaultMessage: "Session Code",
    description: "",
    id: "sessionCode"
  }
};

export default messages;
