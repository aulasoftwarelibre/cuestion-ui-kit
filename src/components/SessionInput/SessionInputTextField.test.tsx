import { shallow } from "enzyme";
import * as React from "react";
import * as renderer from "react-test-renderer";

import messages from "../../languages/en";
import SessionInputTextField from "./SessionInputTextField";

describe("SessionInputTextField", () => {
  const mockOnHandleSubmit = jest.fn();

  it("should render correctly", () => {
    const component = renderer.create(
      <SessionInputTextField
        handleOnSubmit={mockOnHandleSubmit}
        length={4}
        messageEnter={messages.pressEnter}
        messageEnterLengthCharactersCode={messages.enterLengthCharactersCode}
        messageSessionCode={messages.sessionCode}
      />
    ).toJSON();

    expect(component).toMatchSnapshot();
  });

  it("should tell to press ENTER", () => {
    const wrapper = shallow(
      <SessionInputTextField
        handleOnSubmit={mockOnHandleSubmit}
        length={4}
        messageEnter={messages.pressEnter}
        messageEnterLengthCharactersCode={messages.enterLengthCharactersCode}
        messageSessionCode={messages.sessionCode}
      />
    );
    wrapper.find("#session-code-input").simulate("change", {
      target: { name: "session-code-input", value: "1234" }
    });

    expect(
      wrapper.find("#session-code-input").prop("helperText")
    ).toStrictEqual(messages.pressEnter);
  });

  it("should tell to wait until fill code", () => {
    const wrapper = shallow(
      <SessionInputTextField
        handleOnSubmit={mockOnHandleSubmit}
        length={6}
        messageEnter={messages.pressEnter}
        messageEnterLengthCharactersCode={messages.enterLengthCharactersCode}
        messageSessionCode={messages.sessionCode}
      />
    );
    wrapper.find("#session-code-input").simulate("change", {
      target: { name: "session-code-input", value: "1234" }
    });

    expect(
      wrapper.find("#session-code-input").prop("helperText")
    ).toStrictEqual(messages.enterLengthCharactersCode);
  });
});
