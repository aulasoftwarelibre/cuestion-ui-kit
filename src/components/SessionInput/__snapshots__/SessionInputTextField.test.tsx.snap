// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SessionInputTextField should render correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="MuiFormControl-root MuiTextField-root"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
      data-shrink={false}
      htmlFor="session-code-input"
    >
      Session Code
    </label>
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
      onClick={[Function]}
    >
      <fieldset
        aria-hidden={true}
        className="PrivateNotchedOutline-root-57 MuiOutlinedInput-notchedOutline"
        style={
          Object {
            "paddingLeft": 8,
          }
        }
      >
        <legend
          className="PrivateNotchedOutline-legend-58"
          style={
            Object {
              "width": 0.01,
            }
          }
        >
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "&#8203;",
              }
            }
          />
        </legend>
      </fieldset>
      <input
        aria-describedby="session-code-input-helper-text"
        aria-invalid={false}
        className="MuiInputBase-input MuiOutlinedInput-input"
        disabled={false}
        id="session-code-input"
        name="code"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
        value=""
      />
    </div>
    <p
      className="MuiFormHelperText-root MuiFormHelperText-contained"
      id="session-code-input-helper-text"
    >
      Enter {length} characters code.
    </p>
  </div>
</form>
`;
